hw/ip/otbn/data/otbn_testplan.hjson

```diff
@@ -3,30 +3,68 @@
 // SPDX-License-Identifier: Apache-2.0
 {
   name: "otbn"
-  // TODO: remove the common testplans if not applicable
   import_testplans: ["hw/dv/tools/dvsim/testplans/csr_testplan.hjson",
                      "hw/dv/tools/dvsim/testplans/mem_testplan.hjson",
                      "hw/dv/tools/dvsim/testplans/intr_test_testplan.hjson",
                      "hw/dv/tools/dvsim/testplans/tl_device_access_types_testplan.hjson"]
   entries: [
     {
-      name: sanity
+      name: smoke
       desc: '''
-            Basic sanity test acessing a major datapath within the otbn.
+            Smoke test, running a single fixed binary
 
-            **Stimulus**:
-            - TBD
+            This runs the binary from otbn/dv/smoke/smoke_test.s, which is
+            designed to check most of the implemented instructions.
 
-            **Checks**:
-            - TBD
             '''
       milestone: V1
-      tests: ["otbn_sanity"]
+      tests: []
     }
     {
-      name: feature1
-      desc: '''Add more test entries here like above.'''
+      name: single_binary
+      desc: '''
+            Run a single randomly-chosen binary
+
+            This test drives the main bulk of OTBN testing. It picks a random
+            binary from a pre-generated set and runs it, comparing against the
+            model. We'll run this with a large number of seeds and use
+            functional coverage to track when verification of the internals of
+            the core is done.
+
+            Sometimes enable the "done" interrupt to check that it and the
+            error interrupt work correctly.
+
+            '''
       milestone: V1
+      tests: ["otbn_sanity"]
+    }
+
+    {
+      name: reset_recovery
+      desc: '''
+            Run two binaries, resetting the first at an arbitrary time
+
+            This checks that all state is properly re-initialized after a
+            reset. We'd expect X-propagation checks to catch most problems like
+            this, but an explicit reset sequence also adds the relevant
+            FSM/toggle coverage.
```

Comment:
Is this test also intended to check that all state is properly cleared at a reset? 

And what kind of reset are you thinking about? The `rst_ni` input signal, or some form of higher-level reset?

Reply:
Here, I mean the `rst_ni` signal; I'll expand the text slightly to make that clearer.

When you say "check that all state is properly cleared", do you mean a test that grubs around in internal signals to check they are all zero? I wasn't planning something like that: do you think we need it? (Note that we definitely don't clear SRAM on reset, so I'm unconvinced that register contents are a security risk)

Reply:
We did have a discussion with @cdgori in the past about the necessity to clear all internal state (especially non-architecturally-visible state) to ensure that no private key material is left somewhere within OTBN between operations. So it would be a good thing to test that all internal state is really cleared when asserting the rst_ni line.

A full reset (rst_ni) is the most radical reset and should really clear everything. We might, additionally, add mechanisms to trigger such a clear from software. We haven't specified that yet, so there's no need to add a test plan for that feature yet.

Good point about the reset of the SRAM. We haven't discussed that much yet, but it's likely we'll want some clearing operation in hardware as well (as we currently discuss for Ibex). I filed https://github.com/lowRISC/opentitan/issues/4133 to keep track of it (no need to discuss that or find a solution right now).